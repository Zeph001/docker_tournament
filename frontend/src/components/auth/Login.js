import React, { Component } from "react";
import axios from "axios";
import "bootstrap/dist/css/bootstrap-reboot.min.css";
import "../../css/App.css";
import Registration from "./Registration";
import {
  Button,
  Form,
  Grid,
  Header,
  Segment,
} from "semantic-ui-react";
import { Link } from "react-router-dom";

export default class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: "",
      password: "",
      loginErrors: "",
    };

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleSubmit(event) {
    const { email, password } = this.state;

    axios
      .post(
        "http://localhost:3001/sessions",
        {
          user: {
            email: email,
            password: password,
          },
        },
        { withCredentials: true } //that tells the API that it is OK to set the cookie on our client
      )
      .then((response) => {
        if (response.data.status) {
          this.props.handleSuccessfulAuth(response.data);
          this.props.history.push("/")
        }
      })
      .catch((error) => {
        console.log("Login error", error);
      });
    event.preventDefault();
  }

  handleChange(event) {
    this.setState({
      [event.target.name]: event.target.value,
    });
  }

  switchToRegister() {
    return(
        <Registration handleSuccessfulAuth ={this.handleSuccessfulAuth}/>,
      document.getElementById('newRegister')
    )
  }

  render() {
    return (
      <div id="div1">
        <Grid
          textAlign="center"
          style={{ height: "100vh" }}
          verticalAlign="middle"
        >
          <Grid.Column style={{ maxWidth: 450 }}>
            <Header as="h2" color="green" textAlign="center">
              Log-in to your account
            </Header>
            <Form size="large" onSubmit={this.handleSubmit}>
              <Segment stacked>
                  <Form.Input
                    fluid
                    data-testid = "email-input"
                    icon="user"
                    id ="userEmail"
                    iconPosition="left"
                    placeholder="E-mail address"
                    name="email"
                    value={this.state.email}
                    onChange={this.handleChange}
                    required
                    autoComplete="username"
                  />
                  <Form.Input
                    icon="lock"
                    data-testid = "password-input"
                    iconPosition="left"
                    placeholder="Password"
                    type="password"
                    name="password"
                    value={this.state.password}
                    onChange={this.handleChange}
                    required
                    autoComplete="current-password"
                  />
                <Button id ="loginButton" color="green" fluid size="large">
                  Login
                </Button>
              </Segment>
            </Form>
            <Button id= "to-register" color="grey" fluid size="small" attached="top" as={Link} to ="/register">
              New to us?
            </Button>
          </Grid.Column>
        </Grid>
      </div>
    );
  }
}
