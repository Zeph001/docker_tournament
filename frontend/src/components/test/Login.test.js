import Login from "../auth/Login";
import React, { Component } from "react";
import ReactDOM from 'react-dom'
import {render, fireEvent} from '@testing-library/react'
import {BrowserRouter} from "react-router-dom";
import renderer from 'react-test-renderer'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import {shallow, configure, mount} from 'enzyme'

configure({ adapter: new Adapter() });

let wrapper;
beforeEach(() => {
  wrapper = mount(shallow(<BrowserRouter> <Login/> </BrowserRouter>).get(0))
});

afterEach(() => {
  jest.clearAllMocks();
});

test("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(
    <BrowserRouter>
    <Login/>
    </BrowserRouter>, div)
})

//Snapshot is to make sure that the UI doesn't change unexpectedly. It takes a snapshot and compares it to a reference
//If the UI gets changed the snapshot needs to be updated
test("matches the snapshot", () =>{
  const tree = renderer
    .create(
      <BrowserRouter>
        <Login/>
      </BrowserRouter>
    )
    .toJSON();
  console.log(tree)
  expect(tree).toMatchSnapshot();
})

describe('simulate the input fields', () =>{
  test("tests the change in the email input field", () =>{
    const {getByTestId} = render(<BrowserRouter> <Login /> </BrowserRouter>);
    const content = getByTestId("email-input");
    const contentInput = content.children[0]; //Semantic-ui is wrapping the input in a div, so I have to take the child of that div
    fireEvent.change(contentInput, {
      target: { value: "test@test.de" }
    });
    expect(contentInput.value).toEqual("test@test.de")
  })

  test("tests the change in the password input field", () =>{
    const {getByTestId} = render(<BrowserRouter> <Login /> </BrowserRouter>);
    const content = getByTestId("password-input");
    const contentInput = content.children[0];
    fireEvent.change(contentInput, {
      target: { value: "12345678" }
    });
    expect(contentInput.value).toEqual("12345678")
  })

})


  test('validates model on button click', () => {
    const mockCallBack = jest.fn();

    const button = shallow(<Login/>);
    button.find('#loginButton').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(0);
  });
