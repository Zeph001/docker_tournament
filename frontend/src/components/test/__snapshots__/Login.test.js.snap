// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  id="div1"
>
  <div
    className="ui center aligned middle aligned grid"
    style={
      Object {
        "height": "100vh",
      }
    }
  >
    <div
      className="column"
      style={
        Object {
          "maxWidth": 450,
        }
      }
    >
      <h2
        className="ui green center aligned header"
      >
        Log-in to your account
      </h2>
      <form
        className="ui large form"
        onSubmit={[Function]}
      >
        <div
          className="ui stacked segment"
        >
          <div
            className="required field"
          >
            <div
              className="ui fluid left icon input"
              data-testid="email-input"
            >
              <input
                aria-describedby={null}
                autoComplete="username"
                id="userEmail"
                name="email"
                onChange={[Function]}
                placeholder="E-mail address"
                required={true}
                type="text"
                value=""
              />
              <i
                aria-hidden="true"
                className="user icon"
                onClick={[Function]}
              />
            </div>
          </div>
          <div
            className="required field"
          >
            <div
              className="ui left icon input"
              data-testid="password-input"
            >
              <input
                aria-describedby={null}
                autoComplete="current-password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
                value=""
              />
              <i
                aria-hidden="true"
                className="lock icon"
                onClick={[Function]}
              />
            </div>
          </div>
          <button
            className="ui green large fluid button"
            id="loginButton"
            onClick={[Function]}
          >
            Login
          </button>
        </div>
      </form>
      <a
        className="ui grey small fluid top attached button"
        href="/register"
        id="to-register"
        onClick={[Function]}
        role="button"
      >
        New to us?
      </a>
    </div>
  </div>
</div>
`;
